spec/requests/api/shared_language_pages.rb

```diff
@@ -0,0 +1,18 @@
+RSpec.shared_context "shared language pages" do
+
+  let!(:german) { create :language, :german }
+  let!(:french) { create :language, :french }
+  let!(:english) { create :language, :english }
+  let!(:spanish) { create :language, :spanish }
+
+  before do
+    @page_hash = {}
```

Comment:
`Enumerable#inject` is more declarative, and perfect for these situations (no need to change, just mentioning it):

```ruby
before do
  @page_hash = Language.all.inject({}) do |languages, language|
    languages[language.code.to_sym] = { etc ...}
    languages # return languages
  end
end
```